src/components/TrialBanner.js

```diff
@@ -0,0 +1,179 @@
+import _ from 'lodash/fp'
+import { div, h, a, span } from 'react-hyperscript-helpers'
+import { authStore, refreshTerraProfile } from 'src/libs/auth'
+import colors from 'src/libs/colors'
+import { buttonPrimary, Clickable, LabeledCheckbox } from 'src/components/common'
+import { icon, spinner } from 'src/components/icons'
+import { ajaxCaller } from 'src/libs/ajax'
+import { reportError } from 'src/libs/error'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+import Modal from 'src/components/Modal'
+import FreeTrialEulas from 'src/components/FreeTrialEulas'
+
+
+const messages =
+  {
+    'Enabled': {
+      'title': 'Welcome to Terra!',
+      'message': 'You have free compute and storage credits available to upload your data and launch analyses.',
+      'isWarning': false,
+      'enabledLink': {
+        'label': 'Learn more',
+        'url': 'https://software.broadinstitute.org/firecloud/documentation/freecredits'
+      },
+      'button': {
+        'label': 'Start trial',
+        'isExternal': false
+      }
+    },
+    'Enrolled': {
+      'title': 'Access Free Credits',
+      'message': 'You currently have access to your free credits. Learn how to use Terra, about this free credit period, and transitioning to your own billing account once the free credits have expired.',
+      'isWarning': false,
+      'button': {
+        'label': 'Learn More',
+        'url': 'https://software.broadinstitute.org/firecloud/documentation/freecredits',
+        'isExternal': true
+      }
+    },
+    'Terminated': {
+      'title': 'Your free credits have expired',
+      'message': 'Your data will be stored for 30 days from credit expiration date. Learn how you can create your own Google Billing Account and move your data to continue working.',
+      'isWarning': true,
+      'button': {
+        'label': 'Learn more',
+        'url': 'https://software.broadinstitute.org/firecloud/documentation/freecredits?page=faqs',
+        'isExternal': true
+      }
+    }
+  }
+
+export const TrialBanner = _.flow(
+  ajaxCaller,
+  Utils.connectAtom(authStore, 'authState')
+)(class TrialBanner extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      accessingCredits: false,
+      pageTwo: false,
+      termsAgreed: 'false',
+      cloudTermsAgreed: 'false',
+      finalizeTrial: false
+    }
+  }
+
+  render() {
+    const { authState: { isSignedIn, profile }, ajax: { User } } = _.omit('isVisible', this.props)
+    const { accessingCredits, pageTwo, termsAgreed, cloudTermsAgreed, loading, finalizeTrial } = this.state
+    const { trialState } = profile
+    if (!trialState || !isSignedIn || trialState === 'Finalized') return null
+    const { [trialState]: { title, message, enabledLink, button, isWarning } } = messages
+
+    const freeCreditModal = h(Modal, {
```

Comment:
Probably a nit: Is there a cost to creating this modal all the time when it's only going to be used when a user is accepting the free credits? Could be short-circuited by inlining `freeCreditModal` or making it a function that is only called when we need to render the modal.

Reply:
Ooo, interesting point. I had initially pulled it out so that the return statement wasn't so cluttered, but I hadn't thought about how it would then be creating it all the time. I'm not sure if the cost is worth the potential readability of pulling it out, or if the best course of action would be to call it as a function...

My current thought is the function might be the best option in terms of preserving readability and also only being called when necessary, but if others have different opinions, please weigh in!